x-pack/plugins/apm/server/routes/traces/get_aggregated_critical_path.ts

```diff
@@ -106,7 +106,17 @@ export async function getAggregatedCriticalPath({
             map_script: {
               source: `
                 String toHash (def item) {
-                  return item.toString();
+                  long FNV_32_INIT = 0x811c9dc5L;
+                  long FNV_32_PRIME = 0x01000193L;
+                  char[] chars = item.toString().toCharArray();
+                  long rv = FNV_32_INIT;
+                  int len = chars.length;
+                  for(int i = 0; i < len; i++) {
+                      byte bt = (byte) chars[i];
+                      rv ^= bt;
+                      rv *= FNV_32_PRIME;
+                  }
+                  return rv.toString();
                 }
```

Comment:
Wow. Can we add an (api) test for this?

Reply:
There are API tests, including one that tests collisions, in the sense that there is a collision of you use Java's built-in `.hashCode()`. Don't think there's much more that we can do than that.